hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestPrefetch.java

```diff
@@ -298,6 +318,47 @@ public void testPrefetchDoesntSkipRefs() throws Exception {
     });
   }
 
+  @Test
+  public void testOnConfigurationChange() {
+    conf.setInt(PREFETCH_DELAY, 40000);
+    prefetchExecutorNotifier.onConfigurationChange(conf);
+    assertEquals(prefetchExecutorNotifier.getPrefetchDelay(), 40000);
+
+    // restore
+    conf.setInt(PREFETCH_DELAY, 30000);
+    prefetchExecutorNotifier.onConfigurationChange(conf);
+    assertEquals(prefetchExecutorNotifier.getPrefetchDelay(), 30000);
+  }
+
+  @Test
+  public void testPrefetchWithDelay() throws Exception {
+    conf.setInt(PREFETCH_DELAY, 60000);
+    prefetchExecutorNotifier.onConfigurationChange(conf);
+    long totalCompletedBefore = PrefetchExecutor.getPrefetchFutures().size();
+
+    HFileContext context = new HFileContextBuilder().withCompression(Compression.Algorithm.GZ)
+      .withBlockSize(DATA_BLOCK_SIZE).build();
+    Path storeFile = writeStoreFile("TestPrefetchWithDelay", context);
+    setComputeTiming();
+    readStoreFile(storeFile);
+    assertTrue("Elapsed Time {} | Computed Prefetch Delay {}"
+        + getElapsedTime() + prefetchExecutorNotifier.getComputedPrefetchDelay(),
+      getElapsedTime() >= prefetchExecutorNotifier.getComputedPrefetchDelay()
```

Comment:
I don't think this test is correct, as you call your endTimer function only after prefetch is guaranteed to be completed, it may yield true even if the start delay wasn't respected, but the prefetch execution time took longer to complete. 

What I would like to assert here is that right after creating a reader, even if we wait for 5 or 10 secs, no prefetch is running as we set the delay to 60 secs.